staging/src/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go

```diff
@@ -116,6 +116,22 @@ func WithImpersonation(handler http.Handler, requestContextMapper request.Reques
 			}
 		}
 
+		if username != user.Anonymous {
+			// When impersonating a non-anonymous user,
+			// if neither the system:authenticated nor system:unauthenticated groups are explicitly included,
+			// include the system:authenticated group in the impersonated user info
+			found := false
+			for _, group := range groups {
+				if group == user.AllAuthenticated || group == user.AllUnauthenticated {
+					found = true
+					break
+				}
+			}
+			if !found {
+				groups = append(groups, user.AllAuthenticated)
```

Comment:
If any groups have been specified, I don't think that we should auto-add something else to the list.  The caller knew how to send the groups and chose not to.